packages/docusaurus/src/client/normalizeLocation.ts

```diff
@@ -20,7 +20,7 @@ function normalizeLocation<T extends Location>(location: T): T {
 
   let pathname = location.pathname || '/';
   pathname = pathname.trim().replace(/\/index\.html$/, '');
-
+  pathname = pathname.replace(/\/$/, '');
```

Comment:
unfortunately, we can't simply remove a trailing slash like that, as some users might like to have trailing slashes

Reply:
I would prefer that they always had trailing slashes, because then it is clear it is a 'folder' rather than a 'file'.
But surely internally docusaurus needs to have a single normalized version to stop the very problem we are having.
Having said that, I don't know the difference between the location and the url properties, as the url always stays consistent, while the location is the fickle one. If the links were always based on the url property, perhaps all would be well.
I had a quick look at packages/docusaurus/lib/client/exports/Link.js this morning and have now decided to run away for the time being :-)
Thanks for being gentle with me.

Reply:
Thanks for understanding.

These problems are annoying to solve because there are a lot of moving parts and it can impact the SEO of many docs production websites so it is a risky change that should be carefully merged.

Going to close this PR and handle a fix myself when I have more time to work on this